backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/CAdapterGenerator.kt

```diff
@@ -455,6 +456,8 @@ private class ExportedElement(val kind: ElementKind,
                     "result", cfunction[0].second, Direction.KOTLIN_TO_C, builder)
             builder.append("  return $result;\n")
         }
+        builder.append("   } catch (KExceptionObjHolder& e) { TerminateWithUnhandledException(e.obj()); } \n")
```

Comment:
How does this work if nobody throws `KExceptionObjHolder`?

Reply:
Maybe due to type deduplication. Anyway, made it consistent.

Reply:
Actually the test (`produce_dynamic`) passes without this try-catch. And I can't reproduce original issue on macOS neither with nor without this try-catch.
Is the issue Windows-specific? Were you able to reproduce the issue and ensure that it is fixed by this patch?